### ISSUE TITLE
aws-sdk-go Release v1.35.34 added BatchExecuteStatement which is missing in aws-dax-go leading to build failure

### ISSUE THREAD

**Issue by jackdbernier** [aws/aws-dax-go#30]
See https://github.com/aws/aws-sdk-go/pull/3650
This causes build to fail.
```
vendor/github.com/aws/aws-dax-go/dax/service.go:150:5: cannot use (*Dax)(nil) (type *Dax) as type dynamodbiface.DynamoDBAPI in assignment:
	*Dax does not implement dynamodbiface.DynamoDBAPI (missing BatchExecuteStatement method)
```

May I recommend that you move https://github.com/aws/aws-dax-go/blob/master/dax/service.go#L150 to a test instead?
Our code does not rely on `dynamodbiface.DynamoDBAPI`. Without that line our code would build properly since we rely on our own interface.

**Comment by kevinchristen** [comment]
We are aware of the build issue and will have a fix out shortly. I'll discuss your suggestion about not implementing DynamoDBAPI with the team. Thanks for your feedback.

**Comment by kevinchristen** [comment]
Fixed in v1.2.3 and v1.2.4.
